pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -275,7 +273,7 @@ func (s *Service) handle(msg *message, aEvent chan<- service.DIDCommAction) erro
 		s.sendMsgEvents(&service.StateMsg{
 			ProtocolName: DIDExchange,
 			Type:         service.PostState,
-			Msg:          msg.Msg.Clone(),
```

Comment:
It's interesting that we now lose this aspect (no longer exporting a copy).

Reply:
An interface or a struct is fine by me personally. Gophers would probably prefer it be a struct though.

Reply:
An interface should only be returned if that is need to satisfy the signature of another interface.
Otherwise return a struct. The receiver should use an interface, of course.

Reply:
Two public artifacts is overkill in this case. Either make it a public struct or a public interface + private struct. It's immutable and the method set [ID(), Type(), ThreadID(), Scan()] to me strikes the ideal balance between size, cohesion, and generality/usefulness.